ios/browser/api/ledger/brave_ledger.mm

```diff
@@ -108,10 +123,15 @@ typedef NS_ENUM(NSInteger, BATLedgerDatabaseMigrationType) {
 };
 
 @interface BraveLedger () <LedgerClientBridge> {
-  LedgerClientIOS* ledgerClient;
-  ledger::Ledger* ledger;
+  // DO NOT ACCESS DIRECTLY, use `postLedgerTask` or ensure you are accessing
+  // _ledger from a task posted in `_ledgerTaskRunner`
+  std::unique_ptr<ledger::LedgerImpl, ledger::task_deleter<ledger::LedgerImpl>>
```

Comment:
Does this need to be a `LedgerImpl` pointer? Is it possible to avoid the `postLedgerTask` calls by creating a mojo remote that lives in the same process?

Reply:
Nope not possible because of `[Sync]` usages in mojom + ledger at the moment. `[Sync]` type calls are not allowed on the main thread on iOS and iOS is single-process only, so we can't spawn the whole implementation in a separate process to ensure it never uses main